contrib/pam_zfs_key/pam_zfs_key.c

```diff
@@ -756,11 +760,7 @@ pam_sm_open_session(pam_handle_t *pamh, int flags,
 		return (PAM_SUCCESS);
 	}
 
-	int counter = zfs_key_config_modify_session_counter(pamh, &config, 1);
-	if (counter != 1) {
-		zfs_key_config_free(&config);
-		return (PAM_SUCCESS);
-	}
+	(void) zfs_key_config_modify_session_counter(pamh, &config, 1);
 
 	const pw_password_t *token = pw_get(pamh);
```

Comment:
So do I read this correctly that we ask pam for the password even if the dataset is already mounted and even after the first session is already opened? So if e.g. root uses `su` to change to a user they will be prompted for a password even if that user's home is already unlocked?

Reply:
You are correct on both counts.  Even before this change, people were experiencing this problem on the first session (see #11226).  But, of course, once you exit all sessions the next one becomes "first" again; thus, even without this patch, we would expect many `su` users to run into this repetitively.  Indeed, admins who only do one session for a user at a time (i.e. they `su`, and then `exit` that `su` before doing another `su`) might notice absolutely no change in behavior from this patch.

Reply:
Actually I would like to have the "is_mounted" check before we are requesting the password from pam. Because when opening up sessions from `root` via `su` for another user that's already unlocked, there is no need to ask for the password again, and asking for the password and without needing it seems wired to me.

Reply:
I think a larger discussion on the module needs to occur vis-à-vis #11222.  There are multiple possible solutions: it may involve not prompting in the `session` interface, adding a configuration item regarding password prompts, doing as you suggest, or something else entirely.

For the best solution to #11222 (and #11226), we should look at the use cases and prior art (e.g. [pam_ecryptfs](https://github.com/dustinkirkland/ecryptfs-utils/blob/master/src/pam_ecryptfs/pam_ecryptfs.c), [pam_mount](https://sourceforge.net/p/pam-mount/pam-mount/ci/master/tree/src/)).  As it stands, I think the workaround I posted in #11222 should suffice.